src/main/java/com/github/manolo8/darkbot/core/utils/pathfinder/PolygonImpl.java

```diff
@@ -0,0 +1,98 @@
+package com.github.manolo8.darkbot.core.utils.pathfinder;
+
+import eu.darkbot.api.game.other.Area;
+import eu.darkbot.api.game.other.Locatable;
+import eu.darkbot.api.utils.PathFinder;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+public class PolygonImpl extends AreaImpl implements Area.Polygon {
+
+    private final List<? extends Locatable> vertices;
+    private final RectangleImpl bounds = new RectangleImpl();
+
+    private boolean invalidate = true;
```

Comment:
`invalidate` is a verb, `invalid` or `dirty` are nouns

variables should be nouns, while methods should be verbs
```suggestion
    private boolean invalid = true;
```